pep-0646.rst

```diff
@@ -680,6 +680,30 @@ between old unannotated code and new versions of libraries which do use
 these type annotations.
 
 
+Alternatives
+============
+
+It should be noted that the approach outlined in this PEP to solve the
+issue of shape checking in numerical libraries is *not* the only approach
+possible. Examples of lighter-weight alternatives include
+ShapeGuard [#shapeguard]_, tsanley [#tsanley]_, and PyContracts [#pycontracts]_.
+
+While these existing approaches improve significantly on the default
+situation of shape checking only being possible through lengthy and verbose
+assert statements, none of them enable *static* analysis of shape correctness.
+As mentioned in `Motivation`_, this is particularly desirable for
+machine learning applications where, due to library and infrastructure complexity,
+even relatively simple programs must suffer long startup times; iterating
+by running the program until it crashes can be a tedious and frustrating
+experience.
```

Comment:
But don't the tools below improve this experience? Or is the point that those tools still require the program to run through its long startup?

Reply:
Right - ShapeGuard, tsanley and PyContracts are all runtime checkers; essentially glorified asserts. I'll try and clarify this.

Reply:
(Done.)